tests/parsing/TomlConfParserTest.py

```diff
@@ -0,0 +1,316 @@
+aspects = ['aspectname1', 'AspectName2']
+aspectname1.aspect_taste = 80
+aspectname1.subaspect_taste = ['word1', 'word2', 'word3']
+aspectname1.subaspect_taste1 = 'word5'
+
+[foo2]
+inherits = 'foo'
+aspects = 'aspectname3'
+aspectname1.subaspect_taste1 = ['dog', 'cat']
+appends = 'aspectname1.subaspect_taste1'
+a.b.c = '10'
+
+[sample]
+    # coala
+    [sample.item]
+    # Hello World
+    b = [ '1', #Hello
+          '2'
+        ]
+
+    a = 10
+
+[a]
+p = '10'
+q = '20'
+
+[b]
+c = '5'
+d  = '6'
+
+[c]
+inherits = [ 'a', 'b' ]
+p  = 'a'
+d  = 'b'
+appends.a = 'p'
+appends.b = 'd'
+"""
+
+    def setUp(self):
+        self.tempdir = tempfile.gettempdir()
+        self.file = os.path.join(self.tempdir, '.coafile.toml')
+        self.nonexistentfile = os.path.join(self.tempdir, 'w31efr3rk')
+        with open(self.file, 'w') as file:
+            file.write(self.example_file)
+
+        self.uut = TomlConfParser()
+
+        try:
+            os.remove(self.nonexistentfile)
+        except FileNotFoundError:
+            pass
+
+        logger = logging.getLogger()
+
+        with self.assertLogs(logger, 'WARNING') as self.cm:
+            self.sections = self.uut.parse(self.file)
+
+    def tearDown(self):
+        os.remove(self.file)
+
+    def test_parse_nonexisting_file(self):
+        self.assertRaises(FileNotFoundError,
+                          self.uut.parse,
+                          self.nonexistentfile)
+
+    def test_parse_nonexistent_section(self):
+        self.assertRaises(IndexError,
+                          self.uut.get_section,
+                          'non-existent-section'
+                          )
+        self.assertNotEqual(self.uut.parse(self.file, True), self.sections)
+
+    def test_format_value(self):
+        self.assertEqual(self.uut.format_value(80), '80')
+        self.assertEqual(self.uut.format_value(True), 'True')
+        self.assertEqual(self.uut.format_value([1, 2, 3]), '1, 2, 3')
+
+    def test_parse_default_section_deprecated(self):
+        default_should = OrderedDict([
+            ('setting', 'without_section'),
+            ('comment0', 'Hello'),
+            ('(comment0)', '# hello')
+        ])
+
+        key, val = self.sections.popitem(last=False)
+        self.assertTrue(isinstance(val, Section))
+        self.assertEqual(key, 'default')
+
+        is_dict = OrderedDict()
+        for k in val:
+            is_dict[k] = str(val[k])
+        self.assertEqual(is_dict, default_should)
+
+        self.assertRegex(self.cm.output[0],
+                         'A setting does not have a section.')
+
+    def test_parse_all(self):
+        all_should = OrderedDict([
+            ('(comment1)', '# Hello World'),
+            ('max_line_length', '80'),
+            ('ignore', './vendor'),
+            ('a', 'true'),
+            ('(comment2)', '')
+        ])
+
+        # pop off default
+        self.sections.popitem(last=False)
+
+        key, val = self.sections.popitem(last=False)
```

Comment:
If you want only section then first assert that section exists then directly acess that section instead of popping of section one by one